python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1327,11 +1329,22 @@ def get_asset_check_summary_records(
     ) -> Mapping[AssetCheckKey, AssetCheckSummaryRecord]:
         states = {}
         for asset_check_key in asset_check_keys:
-            execution_record = self.get_asset_check_execution_history(asset_check_key, limit=1)
+            last_execution_record = self.get_asset_check_execution_history(asset_check_key, limit=1)
+            last_completed_execution_record = self.get_asset_check_execution_history(
+                asset_check_key, limit=1, status=COMPLETED_ASSET_CHECK_EXECUTION_RECORD_STATUSES
+            )
```

Comment:
we can skip this query if the first fetch returns a completed status check execution

Reply:
yea makes sense